flake8_trio.py

```diff
@@ -34,15 +33,19 @@ def is_trio_call(node: ast.AST, *names: str) -> Optional[str]:
 class Visitor(ast.NodeVisitor):
     def __init__(self) -> None:
         super().__init__()
-        self.problems: list[tuple[int, int]] = []
+        self.problems: List[Tuple[int, int, str]] = []
 
     def visit_With(self, node: ast.With) -> None:
-        for item in node.items:
-            call = is_trio_call(item.context_expr, "fail_after", "move_on_after")
-            if call and not any(isinstance(x, ast.Await) for x in ast.walk(node)):
-                self.problems.append(
-                    (item.lineno, item.col_offset, TRIO100.format(call))
-                )
+
+        # Context manager with no `await` call within
+        calls = [
+            is_trio_call(item.context_expr, "fail_after", "move_on_after")
+            for item in node.items
+        ]
+        if any(calls) and not any(isinstance(x, ast.Await) for x in ast.walk(node)):
+            self.problems.append(
+                (node.lineno, node.col_offset, TRIO100.format(calls.pop()))
+            )
```

Comment:
Actually important to use the `item` attributes as in my previous loop, so that we point at the call to `trio.xxx_after()` if there are multiple clauses e.g.
```python
with (
    trio.move_on_after(10),
    httpx.AsyncContext() as client,
): ...
```

Reply:
`ast.withitem` doesn't have a `lineno` - so it's afaict not possible, changed it because it threw errors at me in tests.

Yeah the Bugbear way sounds good, will look into the boilerplate needed.

Reply:
ah, the `context_expr` in the `withitem` has a lineno though.